assets/js/core.min.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Frontend JavaScript of CMSimple_XH
+ *
+ * <p>
+ * Parts based on Stuart Langridge's serchhi.js
+ * (https://kryogenix.org/code/browser/searchhi/)
+ * </p>
+ *
+ * @file
+ *
+ * @author    The CMSimple_XH developers <devs@cmsimple-xh.org>
+ * @copyright 2018 The CMSimple_XH developers (http://cmsimple-xh.org/?The_Team)
+ * @license   GNU GPLv3 (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ * @since     1.7.3
+ */
+
+var XH = XH || {};
+
+/**
+ * Wraps all occurrences of word inside the given node in a
+ * &lt;span class="xh_find"&gt; element.
+ *
+ * @param {Node}   node
+ * @param {string} word
+ *
+ * @returns {undefined}
+ *
+ * @since 1.7.3
+ */
+XH.highlightSearchWord = function (node, word) {
+    var childNode, startIndex, i;
+
+    for (i = 0; i < node.childNodes.length; i++) {
+        childNode = node.childNodes[i];
+        if (!(childNode.className && childNode.className.match(/\xh_find\b/))) {
+            XH.highlightSearchWord(childNode, word);
+        }
+    }
+    if (node.nodeType === 3 /*Node.TEXT_NODE*/) {
+        startIndex = node.nodeValue.toLowerCase().indexOf(word);
+        if (startIndex !== -1) {
+            XH.doHighlightSearchWord(node, word, startIndex);
+        }
+    }
+};
+
+/**
+ * @private
+ */
+XH.doHighlightSearchWord = function (node, word, startIndex) {
+    var parentNode = node.parentNode;
+    var nodeValue = node.nodeValue;
+    var text1 = nodeValue.substr(0, startIndex);
+    var text2 = nodeValue.substr(startIndex, word.length);
+    var text3 = nodeValue.substr(startIndex + word.length);
+    var hiword = document.createElement("span");
+
+    hiword.className = "xh_find";
+    hiword.appendChild(document.createTextNode(text2));
+
+    parentNode.insertBefore(document.createTextNode(text1), node);
+    parentNode.insertBefore(hiword, node);
+    parentNode.insertBefore(document.createTextNode(text3), node);
+    parentNode.removeChild(node);
+
+};
+
+(function () {
+    var word, i;
+
+    if (XH.searchWords) {
+        for (i = 0; i < XH.searchWords.length; i++) {
+            word = XH.searchWords[i];
+            if (word) {
+                XH.highlightSearchWord(document.body, word.toLowerCase());

```

Comment:
@manu37, regarding commit 49a647532b18ec53a7de3aec7cf0bd30f7d92584: actually, it would be correct to only highlight search results which had been found by the search, i.e. results in the page content. If we knew the page content's container, we could use it here instead of `document.body`.